dashboard/cron.py

```diff
@@ -290,58 +290,137 @@ def update_with_bq_access(self):
 
-            logger.debug(resource_access_df)
+            logger.debug(f'resource_access_df:\n'
+                         f'{resource_access_df}\n'
+                         f'{resource_access_df.dtypes}')
 
-            # Because we're pulling all the data down into one query we need to manipulate it a little bit
-            # Make a copy of the access dataframe
+            # Make resource data from resource_access data
             resource_df = resource_access_df.copy(deep=True)
-            # Drop out the columns user and access time from resource data frame
-            resource_df = resource_df.drop(["user_id", "access_time"], axis=1)
-            # Drop out the duplicates
-            resource_df = resource_df.drop_duplicates(["resource_id"])
+            resource_df = resource_df.drop(
+                columns=['user_id', 'access_time', 'course_id'])
+            resource_df = resource_df.drop_duplicates('resource_id')
+
+            # set resource_id as index for pangres.upsert
+            resource_df: pd.DataFrame = resource_df.set_index('resource_id')
 
-            # Set a dual index for upsert
-            resource_df = resource_df.set_index(["resource_id",])
+            logger.debug(f'resource_df:\n'
+                         f'{resource_df}\n'
+                         f'{resource_df.dtypes}')
 
-            # Drop out the columns resource_type, name from the resource_access
-            resource_access_df = resource_access_df.drop(["resource_type","name"], axis=1)
+            resource_access_df = resource_access_df.drop(
+                columns=['resource_type', 'name'])
 
-            # Drop the columns where there is a Na value
-            resource_access_df_drop_na = resource_access_df.dropna()
+            ra_len_before = len(resource_access_df)
 
-            logger.info(f"{len(resource_access_df) - len(resource_access_df_drop_na)} / {len(resource_access_df)} rows were dropped because of NA")
+            # Drop rows with NA in any column
+            resource_access_df = resource_access_df.dropna()
 
-            # First update the resource table
-            # write to MySQL
+            logger.info(f'{ra_len_before - len(resource_access_df)} / '
+                        f'{ra_len_before} resource_access_df rows with '
+                        'NA values dropped')
+
+            logger.debug(f'resource_access_df:\n'
+                         f'{resource_access_df}\n'
+                         f'{resource_access_df.dtypes}')
+
+            # First, update resource table
             try:
                 dtype = {'resource_id': types.VARCHAR(255)}
-                pangres.upsert(engine=engine, df=resource_df, table_name='resource', if_row_exists='update', dtype=dtype)
+                pangres.upsert(engine=engine, df=resource_df,
+                               table_name='resource', if_row_exists='update',
+                               create_schema=False, add_new_columns=False,
```

Comment:
This line will prevent pangres from adding columns to the table that may be in the dataframe.  It will also prevent creating new schemas.

Reply:
As I've mentioned elsewhere, I opened an issue about this in the pangres repo: ThibTrip/pangres#15